src/sets.jl

```diff
@@ -251,6 +329,14 @@ end
 dual_set(s::SecondOrderCone) = copy(s)
 dual_set_type(::Type{SecondOrderCone}) = SecondOrderCone
 
+function distance_to_set(::DefaultDistance, v::AbstractVector{<:Real}, s::SecondOrderCone)
+    _check_dimension(v, s)
+    t = v[1]
+    xs = v[2:end]
+    result = LinearAlgebra.norm2(xs) - t
+    return max(result, zero(result)) # avoids sqrt
```

Comment:
I personally use the epigraph violation `d = x1 - norm(x[2:end], 2)`, mainly because
* it's easy to compute
* it tells by how much the conic constraint should be "relaxed" for the point to be feasible (where "relaxed" means that you shift the cone's apex)
* if the SOC constraint is written as `x2^2 + ... xn^2 <= x1^2`, the epigraph violation is the square root of this quadratic constraint's absolute violation.

Looks like [COSMO](https://github.com/oxfordcontrol/COSMO.jl/blob/d7913018a8375989fab1d6108be8043288150ea9/src/convexset.jl#L98) uses euclidean distance for computing projections.

Reply:
What about sets like the exponential cone needing domain constraints like `y > 0`?

Reply:
> What is the comment # avoids sqrt for?

legacy comment, just removed. For domain constraints it can be an open question, because some parts of the distance may be non-computable / infinite or complex if the domain is not respected

Reply:
Okay, so for conic sets "Epigraph violation or `Inf` if point fails feasibility bounds" seems like an easily computable and sensible definition.

Reply:
And you call it EpigraphDistance or something like...

Reply:
For norm two distances you can use what proximal solvers do:
https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf page 183 (section 6.3)

Reply:
So, SOC is easy, PSD you just need eigen decomposition, Power and Exponential are well posed even with the extra constraints, but need a Newton...